src/healthcheck.c

```diff
@@ -0,0 +1,422 @@
+		int status;
+		pid_t wait_result = waitpid(pid, &status, 0);
+
+		if (wait_result == -1) {
+			nwarnf("Failed to wait for healthcheck command: %s", strerror(errno));
+			close(stderr_pipe[0]);
+			return false;
+		}
+
+		/* Read stderr output */
+		char stderr_buffer[4096];
+		ssize_t stderr_len = read(stderr_pipe[0], stderr_buffer, sizeof(stderr_buffer) - 1);
+		close(stderr_pipe[0]);
+
+		if (stderr_len > 0) {
+			stderr_buffer[stderr_len] = '\0';
+			/* Trim trailing newlines */
+			while (stderr_len > 0 && (stderr_buffer[stderr_len - 1] == '\n' || stderr_buffer[stderr_len - 1] == '\r')) {
+				stderr_buffer[--stderr_len] = '\0';
+			}
+		} else {
+			stderr_buffer[0] = '\0';
+		}
+
+		if (WIFEXITED(status)) {
+			*exit_code = WEXITSTATUS(status);
+			if (*exit_code != 0) {
+				nwarnf("Healthcheck command failed (exit code %d): %s", *exit_code, config->test[0]);
+				if (stderr_len > 0) {
+					nwarnf("Healthcheck command stderr: %s", stderr_buffer);
+				}
+			}
+			return true;
+		} else if (WIFSIGNALED(status)) {
+			nwarnf("Healthcheck command terminated by signal %d: %s", WTERMSIG(status), config->test[0]);
+			if (stderr_len > 0) {
+				nwarnf("Healthcheck command stderr: %s", stderr_buffer);
+			}
+			*exit_code = 128 + WTERMSIG(status); /* Standard convention for signal termination */
+			return true;
+		} else {
+			nwarnf("Healthcheck command did not terminate normally: %s", config->test[0]);
+			if (stderr_len > 0) {
+				nwarnf("Healthcheck command stderr: %s", stderr_buffer);
+			}
+			*exit_code = -1;
+			return false;
+		}
+	}
+
+	/* This should never be reached */
+	return false;
+}
+
+/* Convert healthcheck status to string */
+const char *healthcheck_status_to_string(int status)
+{
+	if (status >= 0 && status < 4) {
+		return healthcheck_status_strings[status];
+	}
+	return "unknown";
+}
+
+/* Send healthcheck status update to Podman */
+bool healthcheck_send_status_update(const char *container_id, int status, int exit_code)
+{
+	if (container_id == NULL) {
+		return false;
+	}
+
+	const char *status_str = healthcheck_status_to_string(status);
+
+	/* Create simple JSON message for status update */
+	char json_msg[1024];
+	snprintf(json_msg, sizeof(json_msg),
+		 "{\"type\":\"healthcheck_status\",\"container_id\":\"%s\",\"status\":\"%s\",\"exit_code\":%d,\"timestamp\":%ld}",
+		 container_id, status_str, exit_code, time(NULL));
+
+	/* Send via sync pipe to Podman */
+	write_or_close_sync_fd(&sync_pipe_fd, HEALTHCHECK_MSG_STATUS_UPDATE, json_msg);
+
+	return true;
+}
+
+
+/* GLib timer callback function */
+gboolean healthcheck_timer_callback(gpointer user_data)
+{
+	healthcheck_timer_t *timer = (healthcheck_timer_t *)user_data;
+	if (timer == NULL || !timer->timer_active) {
+		return G_SOURCE_REMOVE; /* Stop the timer */
+	}
+
+	/* Check if we're still in start period */
+	if (timer->start_period_remaining > 0) {
+		timer->start_period_remaining -= timer->config.interval;
```

Comment:
This countdown approach creates timing inconsistencies. If the interval is 30s but start period is 45s, the logic becomes unreliable, use actual elapsed time: `time(NULL) - timer->start_time`